test/Tests/src/Semantic/Meta_Spec.enso

```diff
@@ -202,14 +202,17 @@ spec =
         Test.specify "fields of a Type" <|
             typ = Boolean
 
-            Meta.is_atom typ . should_be_true
+            Meta.is_atom typ . should_be_false
             meta_typ = Meta.meta typ
-            meta_typ . should_be_a Meta.Atom
-            fields = case meta_typ of
-                Meta.Atom.Value _ -> meta_typ.constructor.fields
-                _ -> Test.fail "Should be a Meta.Atom.Value: " + meta_typ.to_text
-
-            fields . should_equal []
+            meta_typ . should_be_a Meta.Type
+            cons = case meta_typ of
+                Meta.Type.Value _ -> meta_typ.constructors
+                _ -> Test.fail "Should be a Meta.Type.Value: " + meta_typ.to_text
+
+            cons.length . should_equal 2
+            cons.at 0 . should_be_a Meta.Constructor
```

Comment:
Help experts, please! I am getting:
```
 - [FAILED] fields of a Type [29ms]
        Reason: Expected a value of type Standard.Base.Meta.Constructor but got a value of type Standard.Base.Meta.Atom instead (at enso/test/Tests/src/Semantic/Meta_Spec.enso:213:13-52).
```
I believe I am facing a problem of automatic conversion of zero argument `AtomConstructor` into `Atom` - where `Boolean.True` constructor converts into atom and then wrong  `Meta.Atom` is created.

Can you advice how to fix it? I could write a different test-case without zero argument constructors - and I should do it anyway.

Reply:
Hmm, shouldn't the constructors be wrapped in a `Meta.Constructor.Value`? So this eager evaluation shouldn't really matter. I'm indeed confused.

Anyway, `Boolean` is a bit non-standard type, because you can see that `Meta.meta True` is actually `Meta.Primitive`. Maybe this is actually wrong and it indeed should be `Meta.Atom` since `True` _does pretend_ to be an Atom? cc: @kustosz I think you may have sth to say about it as I remember you were doing stuff to make True/False pretend like they are atoms but be treated as primitives for speed.

Reply:
Please note that [f18b06e](https://github.com/enso-org/enso/pull/5956/commits/f18b06e3d7c5d2f8a2a19d6083b9ed0293fb9166) works OK. E.g. constructors with parameters do work fine.

Reply:
With 1b89cadf0 I've got also the `Boolean`s working. Alas, the code have to be in Java to avoid conversion of zero-argument "constructor" to "atom". Can you explain that to me some day, @kustosz?